src/content/sync/sync-note.ts

```diff
@@ -53,14 +55,39 @@ export async function syncNote(
 
   const existingNoteBlockID = syncedNotes.notes?.[noteItem.key]?.blockID;
 
+  let subContainerBlockID = containerBlockID;
+
   if (existingNoteBlockID) {
+    try {
+      const block = await notion.blocks.retrieve({
+        block_id: existingNoteBlockID,
+      });
+      // get blocks parent
+      if ('parent' in block && block.parent.type === 'block_id') {
+        await notion.blocks.children.list({
+          block_id: block.parent.block_id,
+        });
+        subContainerBlockID = block.parent.block_id;
+      }
+    } catch (error) {
+      if (
+        !isNotionErrorWithCode(error, APIErrorCode.ObjectNotFound) &&
+        !isNotionErrorWithCode(error, APIErrorCode.ValidationError)
+      ) {
+        throw error;
+      }
+    }
```

Comment:
Would it make sense to check the `in_trash` property of the parent instead of relying on the validation error?

```suggestion
    const block = await notion.blocks.retrieve({
      block_id: existingNoteBlockID,
    });
    if ('parent' in block && block.parent.type === 'block_id') {
      const parentBlock = await notion.blocks.retrieve({
        block_id: block.parent.block_id,
      });
      if ('in_trash' in parentBlock && !parentBlock.in_trash) {
        subContainerBlockID = block.parent.block_id;
      }
    }
```